compiler/fir/semantics/src/org/jetbrains/kotlin/fir/resolve/dfa/PersistentLogicSystem.kt

```diff
@@ -173,23 +172,6 @@ abstract class PersistentLogicSystem(context: ConeInferenceContext) : LogicSyste
         return aliasedVariablesThatDontChangeAlias
     }
 
-    private fun PersistentFlow.addVariableAliases(
```

Comment:
Why did you inline those functions? I understand that they had only one usage, but they were introduced to make code of `foldFlow` more readable

Reply:
Because I didn't think `addApprovedStatements` and `addVariableAliases` are very different from `approvedTypeStatements += ...` and `for (alias in aliases) { addLocalVariableAlias(...) }`. Note how I've also grouped all variables in a single immutable map update, too.

Reply:
BTW `aliasedVariablesThatDontChangeAlias` is probably a misnomer, since aliases that *don't change* should already be present in `commonFlow`. More correctly that would be something like aliases that happen regardless of which branch is taken:

```
class C(val x: String?)

fun main() {
  val b = C("...")
  val y: String?
  if (condition) {
    y = b.x
  } else {
    y = b.x
  }
  // both branches *change* aliasing information, but the results are equal
  if (y == null) return
  b.x.length // ok
}
```